### DISCUSSION TITLE
Redirect previous page before login

### DISCUSSION THREAD

**Discussion by nhanco** [nuxt-community/auth-module#1558] [Support Requests]
Hi,
I got a problem when make redirect to previous before login. After login on facebook my screen turn to previous page but few second after that return to home ( route "/"). Something wrong in my code ? Please help me, i'm searched a lot of time. But can't find anything. Sorry english is not my first language.
this is my nuxt.config.js

```  auth: {
     plugins: ["~/plugins/auth.js"],
     cookie: {
       prefix: "auth.",
       options: {
         path: "/",
       },
     },
     redirect: {
       callback: "/callback",
       // login: "/",
       logout: "/",
       rewriteRedirects: false,
       fullPathRedirect: true,
     },
     strategies: {
       facebook: {
         clientId: "xxx",
         endpoints: {
           userInfo:
             "https://graph.facebook.com/v2.12/me?fields=about,name,picture{url},email,birthday",
         },
         scope: ["public_profile", "email", "user_birthday"],
       },
     },
   }
```

and this is my plugins/auth.js
```export default function ({ app, store }) {
  app.router.beforeEach((to, from, next) = {
    if (to.name === "callback" && from.name === null) {
      console.log("alo")
      next({ path:"/temp"});
    }
    next();
  });
}
```

**Comment by akr4m** [comment]
``
export default function({ store, redirect, route }) {
    if (!store.state.auth.loggedIn) {
        return redirect(`/login?redirect=${route.path}`)
    }
}
``

add a middleware like this.

**Comment by spocock** [comment]
@nhanco I think it may partly because you have 

`rewriteRedirects: false,`

in the wrong location so you are not disabling rewrites. It returns to your /callback but then redirects to the original guarded page

It should be 

````
     redirect: {
       callback: "/callback",
       // login: "/",
       logout: "/",
     },
     rewriteRedirects: false,
     fullPathRedirect: true,
````
